### PR TITLE
feat(feedback): create deleted attachment activity for screenshots 

### LINKED ISSUE #68907: [User Feedback] Add an 'activity' note to issues when a attachments are deleted
The proposal is:

When an attachment is deleted, save a note to the Activity log. Record who did the deletion, and when.

Something like this:
Image

Some psuedocode is here: https://github.com/getsentry/sentry/pull/68468/files (this code does not 100% work, and needs tests!)

---

Spinning this off from https://github.com/getsentry/sentry/issues/67707

The original request was to show something like this in the UI (we should do something else though):
Image

But I think that's not practical for a few reasons:
- We're re-using the Issues backend here, which just fully deletes images. So changes to the backend preservation model could be complex
- We'd need to update other places in the ui (/issues page) to detect and support when an attachment exists/doesn't exist and update render to match
- It'll look weird if there were many attachments to start, and they're all deleted now

Instead my proposal is simpler, and while it would affect Issues as well, I think it's a beneficial change without the potential for downstream problems.

### PR DESCRIPTION
**Pull request by aliu39**
Fixes https://github.com/getsentry/sentry/issues/68907

Relates to https://github.com/getsentry/sentry/pull/69880 (frontend changes)

### CODE REVIEW

**Review by JoshFerge — APPROVED** [review]

File: tests/sentry/api/endpoints/test_event_attachment_details.py [hunk 1]
```diff
@@ -138,6 +141,35 @@ def test_delete(self):
...
+
+        self.create_attachment(group_id=None)
+        path = f"/api/0/projects/{self.organization.slug}/{self.project.slug}/events/{self.event.event_id}/attachments/{self.attachment.id}/"
+        response = self.client.delete(path)
+        assert response.status_code == 204
+
+        delete_activity = Activity.objects.get(type=ActivityType.DELETED_ATTACHMENT.value)
+        assert delete_activity.project == self.project
+        assert delete_activity.group_id is None
+        assert delete_activity.group is None
```
Comment (michellewzhang) [inline tests/sentry/api/endpoints/test_event_attachment_details.py:156]: what happens if there's no `group_id`? does it not associate with the issue/feedback?

Comment (aliu3ntry) [reply tests/sentry/api/endpoints/test_event_attachment_details.py:156]: Good point - didn't think this through from the frontend perspective. It probably won't display, so maybe we shouldn't create the activity. Do all feedbacks have a group? @JoshFerge

Comment (michellewzhang) [reply tests/sentry/api/endpoints/test_event_attachment_details.py:156]: all feedbacks should be associated with an issue (group) -- the feedback `id` is basically the group/issue `id`

Comment (aliu3ntry) [reply tests/sentry/api/endpoints/test_event_attachment_details.py:156]: Makes sense... and the attachment is created with the same group, right? I'll change it to only create an Activity when group is not None

Comment (michellewzhang) [reply tests/sentry/api/endpoints/test_event_attachment_details.py:156]: makes sense to me!

**Review by michellewzhang — APPROVED** [review]

**Pull request merged**